src/compiler/checker.ts

```diff
@@ -7688,7 +7688,7 @@ namespace ts {
             }
             // In the following we resolve T[K] to the type of the property in T selected by K.
             const apparentObjectType = getApparentType(objectType);
-            if (indexType.flags & TypeFlags.Union && !(indexType.flags & TypeFlags.Primitive)) {
+            if (indexType.flags & TypeFlags.Union && !(indexType.flags & TypeFlags.Boolean)) {

```

Comment:
Could you add a comment explaining that we are avoiding `TypeFlags.Boolean` here for stylistic reasons to ensure that we show `boolean` and not `true` in error messages? Might also want to link to `keyofAndIndexedAccessErrors.errors.txt`, which is the only test that fails (i.e., has slightly different error messages) if this check isn't here.

Reply:
Done